src/aap_eda/api/serializers/eda_credential.py

```diff
@@ -151,6 +151,22 @@ def validate(self, data):
             id=data.get("credential_type_id")
         )
 
+        # Analytics only allows one credential
+        if (
+            credential_type.name
+            in [
+                enums.AnalyticsCredentialType.BASIC,
```

Comment:
@hsong-rh You might want to move this out to the top as a constant and it will come in handy if we need more Singleton credentials.